[CI] NestedSortBuilderTests testRewrite failing

**Build Scans:**
- [elasticsearch-intake #38579 / part1](https://gradle-enterprise.elastic.co/s/yeiuph5dqsxuq)

**Reproduction Line:**
```
./gradlew ":server:test" --tests "org.elasticsearch.search.sort.NestedSortBuilderTests.testRewrite" -Dtests.seed=E19F1557DB2790DF -Dtests.locale=en-JP -Dtests.timezone=Europe/Sofia -Druntime.java=26
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.search.sort.NestedSortBuilderTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testRewrite'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: an empty collection
     but: <[LEAK: resource was not cleaned up before it was garbage-collected.
Recent access records: 
#1:
	in [elasticsearch[testDoExecuteReleasesRegistrationWhenDataNodeFailsAfterChunkStreaming][generic][T#1]][testDoExecuteReleasesRegistrationWhenDataNodeFailsAfterChunkStreaming]
	org.elasticsearch.search.fetch.chunk.FetchPhaseResponseChunk.close(FetchPhaseResponseChunk.java:206)
	org.elasticsearch.core.Releasables.close(Releasables.java:39)
	org.elasticsearch.core.Releasables.closeExpectNoException(Releasables.java:72)
	org.elasticsearch.action.ActionListenerImplementations$2.run(ActionListenerImplementations.java:51)
	org.elasticsearch.action.ActionListenerImplementations$RunAfterActionListener.onResponse(ActionListenerImplementations.java:314)
	org.elasticsearch.action.ActionListener$3.onResponse(ActionListener.java:438)
	org.elasticsearch.search.fetch.chunk.TransportFetchPhaseResponseChunkAction.lambda$processChunk$2(Transp
[truncated]
```

**Issue Reasons:**
- [main] 1 consecutive failures in test testRewrite
- [main] 1 consecutive failures in step part1
- [main] 1 consecutive failures in pipeline elasticsearch-intake

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -370,27 +370,12 @@ tests:
 - class: org.elasticsearch.xpack.transform.transforms.TransformIndexerStateTests
   method: testStopAtCheckpointForThrottledTransform
   issue: https://github.com/elastic/elasticsearch/issues/147716
-- class: org.elasticsearch.search.sort.NestedSortBuilderTests
-  method: testRewrite
-  issue: https://github.com/elastic/elasticsearch/issues/147721
-- class: org.elasticsearch.search.runtime.StringScriptFieldTermQueryTests
-  method: testToString
-  issue: https://github.com/elastic/elasticsearch/issues/147727
-- class: org.elasticsearch.search.runtime.StringScriptFieldTermQueryTests
-  method: testVisit
-  issue: https://github.com/elastic/elasticsearch/issues/147728
 - class: org.elasticsearch.compute.aggregation.blockhash.BlockHashRandomizedTests
   method: testWithCranky {forcePackedHash=false groups=2 maxValuesPerPosition=1 dups=0 allowedTypes=[Basic[elementType=LONG], Basic[elementType=INT]]}
   issue: https://github.com/elastic/elasticsearch/issues/147732
-- class: org.elasticsearch.search.internal.CancellableBulkScorerTests
-  method: testCancellationStopsScoring
-  issue: https://github.com/elastic/elasticsearch/issues/147737
 - class: org.elasticsearch.search.runtime.GeoPointScriptFieldExistsQueryTests
   method: testToString
   issue: https://github.com/elastic/elasticsearch/issues/147739
-- class: org.elasticsearch.search.runtime.StringScriptFieldTermQueryTests
-  method: testEqualsAndHashCode
-  issue: https://github.com/elastic/elasticsearch/issues/147743
 - class: org.elasticsearch.search.suggest.completion.CategoryQueryContextTests
   method: testNullCategoryIsIllegal
   issue: https://github.com/elastic/elasticsearch/issues/147762

